nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/PutKudu.java

```diff
@@ -0,0 +1,120 @@
+
+import org.apache.nifi.annotation.behavior.EventDriven;
+import org.apache.nifi.annotation.behavior.InputRequirement;
+import org.apache.nifi.annotation.behavior.SupportsBatching;
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.serialization.record.Record;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@EventDriven
+@SupportsBatching
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
+@Tags({"put", "database", "NoSQL", "kudu", "HDFS"})
+@CapabilityDescription("Reads records from an incoming FlowFile using the provided Record Reader, and writes those records " +
+        "to the specified Kudu's table. The schema for the table must be provided in the processor properties or from your source." +
+        " If any error occurs while reading records from the input, or writing records to Kudu, the FlowFile will be routed to failure")
+
+public class PutKudu extends AbstractKudu {
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(KUDU_MASTERS);
+        properties.add(TABLE_NAME);
+        properties.add(SKIP_HEAD_LINE);
+        properties.add(RECORD_READER);
+
+        return properties;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        final Set<Relationship> rels = new HashSet<>();
+        rels.add(REL_SUCCESS);
+        rels.add(REL_FAILURE);
+        return rels;
+    }
+
+    @Override
+    protected Insert insertRecordToKudu(KuduTable kuduTable, Record record, List<String> fieldNames) throws IllegalStateException, Exception {
+        Insert insert = kuduTable.newInsert();
+        PartialRow row = insert.getRow();
+        Schema colSchema = kuduTable.getSchema();
+
+        for (String colName : fieldNames) {
+            int colIdx = this.getColumnIndex(colSchema, colName);
+            if (colIdx != -1) {
+                Type colType = colSchema.getColumnByIndex(colIdx).getType();
+
+                switch (colType.getDataType()) {
+                    case BOOL:
+                        row.addBoolean(colIdx, record.getAsBoolean(colName));
+                        break;
+                    case FLOAT:
+                        row.addFloat(colIdx, record.getAsFloat(colName));
+                        break;
+                    case DOUBLE:
+                        row.addDouble(colIdx, record.getAsDouble(colName));
+                        break;
+                    case BINARY:
+                        row.addBinary(colIdx, record.getAsString(colName).getBytes());
+                        break;
+                    case INT8:
+                    case INT16:
```

Comment:
You can cast it from `int` to `short` before calling `addShort(..)`. If the user is using an `int` field to store a value inclusive of a `short` range then it's safe to cast. If the user is storing something outside the range of a `short` then an `Exception` will be thrown. I think this is a better approach since it allows users to utilize the `INT8` and `INT16` column types if they are able to guarantee that they're in contract with those value ranges.